### ISSUE TITLE
Fails on some URLs (bmj.com)

### ISSUE THREAD

**Issue by huyz** [alin23/sidehn#1]
Doesn't seem to work on https://news.ycombinator.com/item?id=39803845

**Comment by alin23** [comment]
Yes, there are many such URLs, it can't possibly work on every web page. PRs are welcome :)

**Comment by huyz** [comment]
Why can't it possibly work on every page?  I didn't look deep enough into the code to make that obvious.  Are you using some hack or are you fighting web-page specific restrictions?

**Comment by alin23** [comment]
I'm modifying the HTML by adding a new node for the sidebar and moving the whole body into an inner node so that it can be shifted to the left of the sidebar. 

If the page has JS that re-renders the page and changes the HTML structure, then the changes of this extension will be removed or may cause odd layout issues.

**Comment by huyz** [comment]
Ah I see. Good to know.

https://bsky.app may be a good first candidate to play with as it may be a common site to link to…

**Comment by alin23** [comment]
Most such dynamic websites react to DOM tree changes and usually re-render the page. I doubt there is any way to make this extension work well with feed sites like Bluesky or Twitter. 

I simply disable the extension on these websites, it doesn't have to work everywhere. It's good enough that it works on blogs, journals and other static websites.

**Comment by slrgt** [comment]
you might want to look at the reddit for comments extension to see how they made it work on every website. This makes it so you can use hackernews as a comment section for the web. https://github.com/slrgt/Reddit-Comments-for-YouTube

the extension works with reddit and Lemmy comments and is especially convenient for youtube videos since youtube comments are pretty bad

**Comment by johnhamilto** [comment]
Fixed this in #3 :)

**Comment by alin23** [comment]
Yep, fixed by @johnhamilto 
The new version is in review by Google and Mozilla
